File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/Rejuvenation.lua
function GetName()
  return "Rejuvenation"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return { TurnTriggered }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint ~= TurnTriggered then
    return
  end

  char:Heal(char:GetMaxHP() * 0.075)
  char:UseMP(-20)
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/Rejuvenation.cs
using ProjectR.Interfaces.Model;

namespace ProjectR.Scripting.Afflictions
{
    public class Rejuvenation : Affliction
    {
        public override string Name { get { return "Rejuvenation"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.TurnTriggered, 
                };
            }
        }

        protected override void OnTurnTriggered(ICharacter character)
        {
            character.Heal(character.MaxHP * .075);
            character.UseMP(-20);
        }
    }
}